sdk/java-sdk-protobuf/src/main/java/kalix/javasdk/JsonSupport.java

```diff
@@ -130,11 +132,27 @@ public static <T> Any encodeJson(T value, String jsonType) {
    * the JSON string as bytes as value and a type URL starting with "json.kalix.io/".
    *
    * @param valueClass The type of class to deserialize the object to, the class must have the
-   *     proper Jackson annotations for deserialization.
+   *                   proper Jackson annotations for deserialization.
+   * @param any        The protobuf Any object to deserialize.
    * @return The decoded object
    * @throws IllegalArgumentException if the given value cannot be decoded to a T
    */
   public static <T> T decodeJson(Class<T> valueClass, Any any) {
+    return decodeJson(valueClass, any, Optional.empty());
+  }
+
+  /**
+   * Decode the given protobuf Any object to an instance of T using Jackson. The object must have
+   * the JSON string as bytes as value and a type URL starting with "json.kalix.io/".
+   *
+   * @param valueClass       The type of class to deserialize the object to, the class must have the
+   *                         proper Jackson annotations for deserialization.
+   * @param any              The protobuf Any object to deserialize.
+   * @param jacksonMigration The optional @{@link JacksonMigration} implementation used for deserialization.
+   * @return The decoded object
+   * @throws IllegalArgumentException if the given value cannot be decoded to a T
+   */
+  public static <T> T decodeJson(Class<T> valueClass, Any any, Optional<? extends JacksonMigration> jacksonMigration) {
```

Comment:
A `jacksonMigration` additional param to pass the migration from the upper level, otherwise I would need to move the @Migration annotation here.

Reply:
Make the signature with Optional param internal and just have one overload with migration and one without, for Java?

Reply:
I thought about this, but then all the invocation would look like:
```
        extractMigration(typeClass)
          .map(JsonSupport.decodeJson(typeClass, any, _))
          .getOrElse(JsonSupport.decodeJson(typeClass, any))
``` which is also smelly.

Reply:
I thought user API for proto would skip the reflection-migration-lookup and be that the user supplies migrations manually, for that it just seems messy that they'd need to deal with optional, but for the reflective may-have-migration use I understand why it needs to be optional.

Reply:
Another thought: if we support annotation-detection of migration also for proto, can we just hide it inside the JsonSupport instead and the user doesn't have to pass anything?

Reply:
"I thought user API for proto would skip the reflection-migration-lookup" - in some sense, it will, because the user will just use the `public static <T> T decodeJson(Class<T> valueClass, Any any) {` method. Although nothing will prevent him to use the signature with migration.

"can we just hide it inside the" - exactly that's my point one method without any extra params. The "only" drawback is that some code-first annotations will be from the `java-sdk-spring` and some (`@Migration`, `@TypeName` ?) from the `java-sdk-protobuf`.